pkg/invoke/invoker.go

```diff
@@ -1070,10 +1070,13 @@ func (i *Invoker) stream(ctx context.Context, c kclient.WithWatch, thread *v1.Th
 	}
 }
 
-func watchThreadAbort(ctx context.Context, c kclient.WithWatch, thread *v1.Thread, cancel context.CancelCauseFunc) {
+func (i *Invoker) watchThreadAbort(ctx context.Context, c kclient.WithWatch, thread *v1.Thread, cancel context.CancelCauseFunc, run *gptscript.Run) {
 	_, _ = wait.For(ctx, c, thread, func(thread *v1.Thread) (bool, error) {
 		if thread.Spec.Abort {
-			cancel(fmt.Errorf("thread was aborted, cancelling run"))
+			if err := i.gptClient.AbortRun(ctx, run); err != nil {
+				return false, err
+			}
+			go timeoutAfter(ctx, cancel, 30*time.Second)
```

Comment:
Can we add a comment about why this is necessary?

Reply:
I asked for the 30 second thing as a fail safe if the proper abort was to not work for some reason. And yeah add a comment about that.

Reply:
added a comment